guides/hack/08-arrays-and-collections/01-introduction.md

```diff
@@ -39,3 +39,35 @@ Keyset\keys(dict['x' => 1]); // keyset['x']
 
 Vec\map(keyset[1, 2], $x ==> $x + 1); // vec[2, 3]
 ```
+
+## Arrays Cheat Sheet
+
+| Operation| `vec`    | `dict`   | `keyset` |
+|----------|----------|----------|----------|
+| Initialize empty                             | `$v = vec[];`                | `$d = dict[];`                 | `$k = keyset[];`               |
+| Literal                                      | `$v = vec[1, 2, 3];`         | `$d = dict['foo' => 1];`       | `$k = keyset['foo', 'bar'];`   |
+| From Another Container*                      | `$v = vec($container);`      | `$d = dict($keyed_container);` | `$k = keyset($container);`     |
+| Keys from Container*                         | `$v = Vec\keys($container);` | N/A                            | `$k = Keyset\keys($container);`|
+| Add Elements                                 | `$v[] = 4;`                  | `$d['baz'] = 2;`               | `$k[] = 'baz';`                |
+| Bulk Add Elements                            | `$v = Vec\concat($t1, $t2)`  | `$d = Dict\merge($kt1, $kt2)`  | `$k = Keyset\union($t1, $t2)`  |
+| Remove Elements                              | Remove-at-index is unsupported; `$first = PHP\array_shift($v)` and `$last = PHP\array_pop($v)` | `unset($d['baz']);`| `unset($k['baz']);`|
+| Key Existence                                | `C\contains_key($v, 1)`      | `C\contains_key($d, 'foo')`    | `C\contains_key($k, 'foo')`    |
+| Value Existence                              | `C\contains($v, 3)`          | `C\contains($d, 2)`            | N/A                            |
```

Comment:
`C\contains()` is also fine for sets, it's just equivalent to `C\contains_key()` - perhaps explicitly note "use `C\contains_key()`" ?

Reply:
![image](https://user-images.githubusercontent.com/5179225/127913353-47f1da67-21bc-4576-ba92-735c7b87067e.png)